ffi/src/primitives.rs

```diff
@@ -133,5 +141,153 @@ primitive_derive! { u32, i32, u64, i64 }
 #[cfg(not(target_family = "wasm"))]
 primitive_derive! { u8, i8, u16, i16 }
 
-// TODO: u128/i128 is not FFI-safe. Must be properly serialized!
-primitive_derive! { u128, i128 }
+/// Ffi-safe representation of [`u128`]
+#[derive(Clone, Copy, Debug, Default)]
+#[repr(transparent)]
+pub struct FfiU128(FfiTuple2<u64, u64>);
+
+// SAFETY: Transparent to `FfiTuple<u64, u64>` which is `ReprC`
+unsafe impl ReprC for FfiU128 {}
+
+/// Ffi-safe representation of [`i128`]
+#[derive(Clone, Copy, Debug, Default)]
+#[repr(transparent)]
+pub struct FfiI128(FfiTuple2<u64, u64>);
+
+// SAFETY: Transparent to `FfiTuple<u64, u64>` which is `ReprC`
+unsafe impl ReprC for FfiI128 {}
```

Comment:
```suggestion
/// Ffi-safe representation of [`u128`]
#[derive(Clone, Copy, Debug, Default)]
#[repr(transparent)]
pub struct FfiU128(FfiTuple2<u64, u64>);

// SAFETY: Transparent to `FfiTuple<u64, u64>` which is `ReprC`
unsafe impl ReprC for FfiU128 {}

/// Ffi-safe representation of [`i128`]
#[derive(Clone, Copy, Debug, Default)]
#[repr(transparent)]
pub struct FfiI128(FfiTuple2<u64, u64>);

// SAFETY: Transparent to `FfiTuple<u64, u64>` which is `ReprC`
unsafe impl ReprC for FfiI128 {}
```
you can put this in a macro as well